UOP1_Project/Assets/Scripts/Audio/AudioManager.cs

```diff
@@ -0,0 +1,37 @@
+﻿using UnityEngine;
+
+public class AudioManager : MonoBehaviour
+{
+    private AudioSource audioSource = default;
+
+    private void Awake()
+    {
+        PreventMultipleAudioManagers();
+        audioSource = GetComponent<AudioSource>();
+    }
+
+    public void Play(AudioClip audioClip)
+    {
+        audioSource.clip = audioClip;
+        audioSource.Play();
+    }
+
+    public void Stop()
+    {
+        audioSource.Stop();
+    }
+
+    private void PreventMultipleAudioManagers()
+    {
+        UnityEngine.Object[] otherAudioManagers = FindObjectsOfType(GetType());
```

Comment:
Shouldn't this be AudioManager[], since casting it as Object would be boxing, and that's bad for performance. 

And it's better for readability to do Audiomanager[].

And I think you're basically trying to create a singleton? you can just do the normal singleton initialization if you want, (have a static instance., check if instance is null)...ect